pinot-broker/src/main/java/org/apache/pinot/broker/routing/v2/segmentselector/SegmentSelector.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.routing.v2.segmentselector;
+
+import java.util.List;
+import java.util.Set;
+import org.apache.helix.model.ExternalView;
+import org.apache.pinot.common.request.BrokerRequest;
+
+
+/**
+ * The segment selector selects the segments for the query. The segments selected should cover the whole dataset (table)
```

Comment:
`The segments selected should the whole dataset (table) without overlap` is a bit vague. This statement naturally makes me think of the selection step for merge/roll up use case where we can have segments with duplicate data within the same table. If I look at the actual `offline, realtime segment selector` implementation, it currently handles "OFFLINE/ONLINE/ERROR/CONSUMING" issues, which is slightly different from `selecting whole dataset`. Current code looks more like `filtering out invalid segments`.

Also, the definition of `whole dataset` can be either `all segments from idealstate` or `all segments in external view with at least one ONLINE` and so on.  Are we considering this as the implementation specific issue for each type of segment selector?

Reply:
Only LLC realtime table looks at the external view CONSUMING state to ensure only the first CONSUMING segment for each partition getting selected to prevent overlap data. It will not look at other states, which will be handled by the instance selector instead.